### ISSUE TITLE
Request - Pressing escape should close the Tearsheet

### ISSUE THREAD

**Issue by nertzy** [carbon-design-system/ibm-security#742]
### Proposal

When I open a Tearsheet, I find myself reaching for the <kbd>Esc</kbd> key to close it.

I propose that pressing <kbd>Esc</kbd> should be equivalent to clicking the close button. It should close the Tearsheet if possible, and if the close button is disabled, it should do nothing.

### Goals / use cases

- [ ] Better keyboard accessibility

**Comment by SimonFinney** [comment]
@petervachon What are your thoughts on the above proposal? Modals and panels already behave similarly

**Comment by petervachon** [comment]
Carbon uses the ESC key for closing modals so that works for me 👍

**Comment by nertzy** [comment]
This is still relevant.

**Comment by SimonFinney** [comment]
Superseded by https://github.com/carbon-design-system/ibm-cloud-cognitive/issues/183
